test/boost/tablets_test.cc

```diff
@@ -2944,6 +2944,45 @@ SEASTAR_THREAD_TEST_CASE(test_load_balancing_merge_colocation_with_single_rack)
     }).get();
 }
 
+// Verify merge can proceed with multiple racks and RF=#racks
+//
+// Given replica sets (not in rack order):
+// rack1 { n1, n2 }
+// rack2 { n3, n4 }
+//
+// t0: { n1, n3 }
+// t1: { n4, n2 }
+//
+SEASTAR_THREAD_TEST_CASE(test_load_balancing_merge_colocation_with_multiple_racks_and_rf_equals_racks) {
+    do_with_cql_env_thread([] (auto& e) {
+        const int rf = 2;
+        const int n_racks = rf;
+        const int n_hosts = 4; // 2 nodes in each rack.
+        const unsigned shard_count = 1;
```

Comment:
why?  It would be good to test this with more shards as well.

Reply:
number of shards should be irrelevant but we must test the colocation picks the right shard on the destination node.  Having a pytest for verifying that is good enough.

Reply:
> number of shards should be irrelevant but we must test the colocation picks the right shard on the destination node. Having a pytest for verifying that is good enough.

If co-location migration is emitted to wrong shard, then our existing tests (and new ones) wouldn't be able to complete merge, since there's an additional pass over replica sets of sibling tablets (with topology locked) to verify they're indeed co-located. It's decoupled from co-location migration planner, see load_balancer::all_sibling_tablet_replicas_colocated().